kong/kong.lua

```diff
@@ -252,6 +252,18 @@ function Kong.balancer()
   end
 end
 
+function Kong.rewrite()
+  core.rewrite.before()
+
+  for _, plugin in ipairs(singletons.loaded_plugins) do
+    if plugin.schema.no_consumer and plugin.schema.no_api then
+      plugin.handler:rewrite()
+    end
+  end
+
+  core.rewrite.after()
+end
+
```

Comment:
the above runs on `loaded_plugins` which means that it will run all plugins available, no matter what. So the user has no control on whether it runs or not, when it is in the system, it will execute. 

It should fetch the plugins from the database, and execute only those that are neither attached to a consumer, nor to an api (eg. the ones that already run on every api).

btw, can't we bring the api detection forward, to run in the `core.rewrite.before()` handler? not sure what it takes, but then we can run plugins like in any phase, including "per api" configured plugins